### PR TITLE
Issue Comment Improvements

### LINKED ISSUE #159: Fix Schizo Bot
Bot shouldn't ask itself questions. Or any questions to the "mannequin" DevPool account.

              @ubiquity-bounties would you please post your wallet address here?

_Originally posted by @ubiquibot[bot] in https://github.com/ubiquity/bounty-bot/issues/98#issuecomment-1453148603_

### PR DESCRIPTION
**Pull request by Steveantor**
Resolves #159 #160

### CODE REVIEW

**Comment by pavlovcik** [comment]
Can we test and prove it works on your repo? I'll look into it next time I'm on your computer but if you can link proof that would be ideal.

**Comment by pavlovcik** [comment]
I just enabled auto-build on forks. Maybe you can test it out by re-enabling the action and getting the latest.

https://github.com/ubiquity/bounty-bot/commit/c27389da60c297d988c34aa02374e0d39d53a53a

File: src/handlers/payout/index.ts [hunk 1]
```diff
@@ -34,8 +35,17 @@ export const handleIssueClosed = async () => {
...
   if (!recipient) {
-    log.info(`Recipient address is missing`);
-    await addCommentToIssue(`@${assignee.login} would you please post your wallet address here?`, issue.number);
+    const {
+      data: { state_reason },
+    } = await context.octokit.issues.get({
+      owner: payload.organization!.login,
+      repo: payload.repository.name,
+      issue_number: issue.number,
+    });
```
Comment (0xcodercrane) [inline src/handlers/payout/index.ts:44]: We have a payload including issue object. Please take a look at the payload object and getBotContext(). We shouldn't get issue object again in the codebase, should use the one provided by Probot (basically both are same)

File: src/handlers/payout/index.ts [hunk 2]
```diff
@@ -34,8 +35,17 @@ export const handleIssueClosed = async () => {
...
+    if (!RESERVED_USERNAMES[assignee.login] && state_reason === "completed") {
```
Comment (0xcodercrane) [inline src/handlers/payout/index.ts:45]: `state_reason === "completed"`

If there's not predefined types for `completed`, we need to define them for issue status and use it here.

**Review by 0xcodercrane — CHANGES_REQUESTED** [review]
Lets just make sure its getting tested properly with Test issue before merging

File: src/handlers/payout/index.ts (hunk 1 above)
Comment (Steveantor) [reply src/handlers/payout/index.ts:44]: the issue object doesn't have the `state_reason` property

File: src/handlers/payout/index.ts (hunk 2 above)
Comment (Steveantor) [reply src/handlers/payout/index.ts:45]: what do u mean by that? `state_reason === "completed"` means that the issue was closed as completed (provided by the GitHub API), 
as oppossed to "not planned"

Comment (0xcodercrane) [reply src/handlers/payout/index.ts:49]: What I mean by that is that we shouldn't use hardcoded values `completed` here. we can define types for `completed` and `not planned` somewhere and use it

**Comment by 0xcodercrane** [comment]
https://github.com/ubiquity/bounty-bot/commit/8dbbf7982f9ab117dad6d271ff73c74d51854dc9

Please take a look at the above commit @Steveantor. I've added state_reason to Issue schema, you should be able to get the state_reason like issue.state_reason and compare it like state_reason == StateReason.COMPLETED

**Comment by Steveantor** [comment]
updated @0xcodercrane

File: src/handlers/payout/index.ts [hunk 3]
```diff
@@ -38,8 +39,10 @@ export const handleIssueClosed = async () => {
   console.log({ assignee });
```
Comment (0xcodercrane) [inline src/handlers/payout/index.ts:39]: it looks like you forget to remove console log lol

**Comment by 0xcodercrane** [comment]
will merge it and make a couple of changes on the development branch if needed

**Review by 0xcodercrane — APPROVED** [review]

File: src/handlers/payout/index.ts (hunk 3 above)
Comment (Steveantor) [reply src/handlers/payout/index.ts:39]: I see haha

**Comment by pavlovcik** [comment]
> Resolves #159 #160

Also for the future: you should only solve one issue per pull request. I had to manually link the issue and if I didn't then the bot would only pay for the first issue mentioned.

File: src/configs/shared.ts [hunk 4]
```diff
@@ -25,3 +25,7 @@ export const DEFAULT_CHAIN_ID = 1;
 export const DEFAULT_RPC_ENDPOINT = "https://rpc-pay.ubq.fi/v1/mainnet";
 export const DEFAULT_PAYMENT_TOKEN = "0x6B175474E89094C44Da98b954EedeAC495271d0F";
 export const DEFAULT_PERMIT_BASE_URL = "https://pay.ubq.fi";
+
+export const RESERVED_USERNAMES: { [index: string]: boolean } = {
+  "ubiquity-bounties": true,
```
Comment (pavlovcik) [inline src/configs/shared.ts:30]: Seems like an unusual implementation. I would probably do an array of strings.

Comment (Steveantor) [reply src/configs/shared.ts:30]: Array is slower for this case, since key:value pairs are O(1), It can be a Set() instead

**Pull request merged**